catroid/src/main/java/org/catrobat/catroid/transfers/GoogleLoginHandler.java

```diff
@@ -47,56 +41,44 @@
 import org.catrobat.catroid.utils.DeviceSettingsProvider;
 import org.catrobat.catroid.utils.ToastUtil;
 
-public class GooglePlusLoginHandler implements GoogleApiClient.ConnectionCallbacks,
-		GoogleApiClient.OnConnectionFailedListener,
-		CheckOAuthTokenTask.OnCheckOAuthTokenCompleteListener,
+import static com.google.android.gms.auth.api.signin.GoogleSignIn.getClient;
+import static com.google.android.gms.auth.api.signin.GoogleSignIn.getSignedInAccountFromIntent;
+
+public class GoogleLoginHandler implements CheckOAuthTokenTask.OnCheckOAuthTokenCompleteListener,
 		GoogleLogInTask.OnGoogleServerLogInCompleteListener,
 		CheckEmailAvailableTask.OnCheckEmailAvailableCompleteListener,
 		GoogleExchangeCodeTask.OnGoogleExchangeCodeCompleteListener {
 
 	private AppCompatActivity activity;
-	public static final int GPLUS_REQUEST_CODE_SIGN_IN = 0;
 	public static final int REQUEST_CODE_GOOGLE_PLUS_SIGNIN = 100;
-	public static final int STATUS_CODE_SIGN_IN_CURRENTLY_IN_PROGRESS = 12502;
-	private GoogleApiClient googleApiClient;
-	private static final String GOOGLE_PLUS_CATROWEB_SERVER_CLIENT_ID = "427226922034-r016ige5kb30q9vflqbt1h0i3arng8u1.apps.googleusercontent.com";
+	private GoogleSignInClient googleSignInClient;
+	private static final String GOOGLE_LOGIN_CATROWEB_SERVER_CLIENT_ID = "427226922034-r016ige5kb30q9vflqbt1h0i3arng8u1.apps.googleusercontent.com";
 
-	public GooglePlusLoginHandler(AppCompatActivity activity) {
+	@SuppressWarnings("RestrictedApi")
```

Comment:
Is there a reason for a SuppressWarnings

Reply:
As seen in this link https://stackoverflow.com/questions/49464246/when-using-firebase-googlesignin-is-marked-as-internal-and-should-not-be-access it is a known issue in version 12.0.0